src/components/MobileMetricCard/SwipeableCard.js

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from 'react'
+import cx from 'classnames'
+import Icon from '@santiment-network/ui/Icon'
+import styles from './SwipeableCard.module.scss'
+
+const BUTTON_WIDTH = 85
+
+const FULL_HIDE_POSITION = -2000
+
+const SwipeableCard = ({
+  children,
+  onLeftActionClick,
+  onRightActionClick,
+  hasLeftAction,
+  hasRightAction = true,
+  hide,
+  isSelected,
+  width
+}) => {
+  const [startPosition, setStartPosition] = useState(0)
+  const [offset, setOffset] = useState(0)
+  const [side, setSide] = useState('right')
+
+  let [currentGesture, setCurrentGesture] = useState(null)
+
+  if (hide && offset !== FULL_HIDE_POSITION) {
+    setOffset(FULL_HIDE_POSITION)
+  }
+
+  if (!hide && offset === FULL_HIDE_POSITION) {
+    setOffset(0)
+  }
+
+  const onStart = evt => {
+    setCurrentGesture({
+      startX: evt.touches[0].pageX,
+      prevX: evt.touches[0].pageX,
+      prevTs: Date.now(),
+      startPosition
+    })
+  }
+
+  const onMove = evt => {
+    if (!currentGesture) {
+      return
+    }
+
+    const { startX, prevX, prevTs, startPosition } = currentGesture
+    const x = evt.touches[0].pageX
+    const dx = x - startX
+    const offset = startPosition + dx
+    if ((offset < 0 && !hasRightAction) || (offset > 0 && !hasLeftAction)) {
+      return
+    }
+
+    setOffset(offset)
+    setSide(offset > 0 ? 'left' : 'right')
+
+    const ts = Date.now()
+    const speed = Math.abs(x - prevX) / (ts - prevTs)
+
+    if ((speed > 4 && ts !== prevTs) || (Math.abs(offset) * 100) / width > 85) {
```

Comment:
I guess it's better to move these checks to the function. Right now, it's hard to understand its purpose on a first glance